fix(k8sprocessor): fix metadata dependencies

astencel-sumo opened the pull request:
Fixes #512

pmalek-sumo commented on pkg/processor/k8sprocessor/kube/owner_test.go:
```diff
@@ -47,6 +48,201 @@ func waitForWatchToBeEstablished(client *fake.Clientset, resource string) <-chan
+			return false
+		}
+
+		if uid := owners[0].UID; uid != "fb9e6935-8936-4959-bd90-4e975a4c2b07" {
```
Maybe it would be more readable to assign the ReplicaSet to a variable and then use its ID here instead of using magic string?
And similarly in other tests that are added with this PR.

WDYT?

astencel-sumo replied:
OK, added variable names.

astencel-sumo commented on pkg/processor/k8sprocessor/kube/client_test.go:
```diff
@@ -40,7 +40,23 @@ import (
 func newFakeAPIClientset(_ k8sconfig.APIConfig) (kubernetes.Interface, error) {
-	return fake.NewSimpleClientset(), nil
+	fakeClient := fake.NewSimpleClientset()
+	// Add batch/v1 CronJob resource so that setting up a CronJob informer works in testing.
```
Added comment.

pmalek-sumo commented:
> One comment w.r.t to the tests is that I'd maybe be more generous with comments, i.e. add comments where we're testing this particular feature that is added in this PR so that we know what's being tested.

ping

astencel-sumo commented:
> > One comment w.r.t to the tests is that I'd maybe be more generous with comments, i.e. add comments where we're testing this particular feature that is added in this PR so that we know what's being tested.
> 
> ping

Here's how I see it: there was a functionality  that was not covered by tests (retrieving metadata for deployments and cronjobs), and this functionality didn't work correctly. I fixed the bug and added the tests for this functionality at the same time. I don't think there is anything else required with regards to this bug.
